pkg/lightning/backend/tidb/tidb.go

```diff
@@ -753,8 +816,27 @@ stmtLoop:
 			err    error
 		)
 		for i := 0; i < writeRowsMaxRetryTimes; i++ {
-			stmt := stmtTask.stmt
-			result, err = be.db.ExecContext(ctx, stmt)
+			query := stmtTask.stmt
+			if be.cachePrepStmts {
+				var prepStmt *sql.Stmt
+				key := &stmtKey{query: query}
+				be.stmtCacheMutex.RLock()
+				stmt, ok := be.stmtCache.Get(key)
+				be.stmtCacheMutex.RUnlock()
+				if ok {
+					prepStmt = stmt.(*sql.Stmt)
+				} else if stmt, err := be.db.Prepare(query); err == nil {
+					prepStmt = stmt
+					be.stmtCacheMutex.Lock()
+					be.stmtCache.Put(key, stmt)
```

Comment:
`execStmts` is called concurrently, this might override existing stmt without close it.

Reply:
yes, cached prepstmt will be concurrent use by multiple goroutines. any idea how to avoid override existing stmt without close it?

Reply:
after obtaining the lock, check again the existence of cache entry.